src/puppetlabs/puppetdb/query/facts.clj

```diff
@@ -70,15 +70,13 @@
   ascending. This includes facts which are known only for deactivated and
   expired nodes."
   ([]
-     (fact-names {}))
+   (fact-names {}))
   ([paging-options]
-     {:post [(map? %)
-             (coll? (:result %))
-             (every? string? (:result %))]}
-     (paging/validate-order-by! [:name] paging-options)
-     (let [facts (query/execute-query
-                  ["SELECT DISTINCT name
-                   FROM fact_paths
-                   ORDER BY name"]
-                  paging-options)]
-       (update-in facts [:result] #(map :name %)))))
+   {:post [(map? %)
+           (coll? (:result %))
+           (every? string? (:result %))]}
+   (paging/validate-order-by! [:name] paging-options)
```

Comment:
probably yes to both, but this is preexisting code unrelated to the patch and we use more or less the same constructions in the same functions in every query namespace. I think a coordinated change would be a better idea in this case too.